kong/plugins/basic-auth/api.lua

```diff
@@ -52,5 +52,29 @@ return {
     DELETE = function(self, dao_factory)
       crud.delete(self.basicauth_credential, dao_factory.basicauth_credentials)
     end
+  },
+  ["/basic-auth/"] = {
+    GET = function(self, dao_factory)
+      crud.paginated_set(self, dao_factory.basicauth_credentials)
+    end
+  },
+  ["/basic-auth/:credential_id/consumer"] = {
```

Comment:
The `/consumers/:username_or_id/basic-auth/:credential_username_or_id` endpoint provided by this plugin allows to retrieve a credential based on its `username` attribute, or its `id`.

Maybe we should allow the same for this endpoint as well? Any reason why we couldn't do so?

```lua
local credentials, err = crud.find_by_id_or_field(dao_factory.basicauth_credentials, nil
                                                  self.params.credential_username_or_id,
                                                  "username")
```